fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/tree/pdagg/PushDownAggregateRewriter.java

```diff
@@ -0,0 +1,443 @@
+            }
+        }
+
+        childContext.origAggregator = context.origAggregator;
+
+        if (join.getOnPredicate() != null) {
+            join.getOnPredicate().getUsedColumns().getStream().filter(childOutput::contains)
+                    .mapToObj(factory::getColumnRef).forEach(c -> childContext.groupBys.put(c, c));
+        }
+
+        if (join.getPredicate() != null) {
+            join.getPredicate().getUsedColumns().getStream().filter(childOutput::contains)
+                    .mapToObj(factory::getColumnRef).forEach(v -> childContext.groupBys.put(v, v));
+        }
+
+        return process(joinOpt.inputAt(child), childContext);
+    }
+
+    @Override
+    public OptExpression visitLogicalCTEAnchor(OptExpression optExpression, AggregateContext context) {
+        optExpression.setChild(0, process(optExpression.inputAt(0), AggregateContext.EMPTY));
+        optExpression.setChild(1, process(optExpression.inputAt(1), context));
+        return optExpression;
+    }
+
+    @Override
+    public OptExpression visitLogicalTableScan(OptExpression optExpression, AggregateContext context) {
+        if (isInvalid(optExpression, context)) {
+            return visit(optExpression, context);
+        }
+
+        if (context.aggregations.isEmpty() && context.groupBys.isEmpty()) {
+            return visit(optExpression, context);
+        }
+
+        // check groupBys is from orig aggregation, not from JoinNode
+        if (!context.groupBys.keySet().stream().allMatch(s -> allPushDownGroupBys.contains(s))) {
+            return visit(optExpression, context);
+        }
+
+        Preconditions.checkState(context.groupBys.values().stream().allMatch(ScalarOperator::isColumnRef));
+
+        LogicalScanOperator scan = (LogicalScanOperator) optExpression.getOp();
+
+        OptExpression result = optExpression;
+        // if the aggregation is complex expression, need create project
+        if (context.aggregations.values().stream().map(c -> c.getChild(0)).anyMatch(s -> !s.isColumnRef())) {
+            Map<ColumnRefOperator, ScalarOperator> refs = Maps.newHashMap();
+            scan.getOutputColumns().forEach(c -> refs.put(c, c));
+
+            for (Map.Entry<ColumnRefOperator, CallOperator> entry : context.aggregations.entrySet()) {
+                ScalarOperator input = entry.getValue().getChild(0);
+                if (!input.isColumnRef()) {
+                    ColumnRefOperator ref = factory.create(input, input.getType(), input.isNullable());
+                    refs.put(ref, input);
+                    entry.getValue().setChild(0, ref);
+                }
+            }
+
+            result = OptExpression.create(new LogicalProjectOperator(refs), result);
+        }
+
+        LogicalAggregationOperator aggregate = new LogicalAggregationOperator(AggType.GLOBAL,
+                Lists.newArrayList(context.groupBys.keySet()), context.aggregations);
```

Comment:
The key of groupBy map is ColumnRefOperator. I think we don't need add the project then agg. We can extract columns from these scalarOperators as group key and add a project on this agg. This may helps to reduce the group key numbers.

Reply:
1. Your example, it's should support in other rule
2. The situation you said has been handled here, add project to handle `select sum(a + b)`, groupBys only save columnRef there